ooniprobe/AppDelegate.mm

```diff
@@ -143,6 +155,53 @@ -(void)application:(UIApplication *)application didReceiveLocalNotification:(UIL
     //TODO start your long running bg task here
 }
 
+- (BOOL)application:(UIApplication *)application handleOpenURL:(NSURL *)url {
+    //NSLog(@"url recieved: %@", url);
+    //NSLog(@"query string: %@", [url query]);
+    
+    //creating parameters dict
+    NSDictionary *dict = [DictionaryUtility parseQueryString:[url query]];
+    NSDictionary *parameters = [DictionaryUtility getParametersFromDict:dict];
+    
+    NSString *minimum_version = [parameters objectForKey:@"mv"];
+    if (minimum_version != nil){
+        if ([minimum_version compare:[[[NSBundle mainBundle] infoDictionary] objectForKey:@"CFBundleShortVersionString"] options:NSNumericSearch] == NSOrderedDescending) {
+            //actualVersion is lower than the requiredVersion
+            UIAlertView *alertView = [[UIAlertView alloc] initWithTitle:NSLocalizedString(@"ooniprobe_outdate", nil) message:NSLocalizedString(@"ooniprobe_outdate_msg", nil) delegate:nil cancelButtonTitle:NSLocalizedString(@"Cancel", nil) otherButtonTitles:NSLocalizedString(@"ok", nil), nil];
+            alertView.tag = 2;
+            [alertView show];
+        }
+        else {
+            NSString *action = [url host];
+            if ([action isEqualToString:@"nettest"]){
+                //For now checking only test name
+                if ([parameters objectForKey:@"tn"] && [[Tests currentTests] getTestWithName:[parameters objectForKey:@"tn"]])
+                    [self openURIschemeScreen:parameters];
+                else {
+                    UIAlertView *alertView = [[UIAlertView alloc] initWithTitle:NSLocalizedString(@"invalid_parameter", nil) message:[NSString stringWithFormat:@"%@ : %@", NSLocalizedString(@"test_name", nil), [parameters objectForKey:@"tn"]] delegate:nil cancelButtonTitle:NSLocalizedString(@"Cancel", nil) otherButtonTitles:nil];
```

Comment:
For reading the diffs (especially in the terminal) it may be better to wrap very long lines.

Reply:
@lorenzoPrimi do you know whether it's possible to configure Xcode to reformat the code (or can we use directly `clang-format` for such purpose)? After this release cycle I'd like to reformat the code a bit to have shorter lines.

Reply:
I checked, there is clearly support for objc in `clang-format`. I think we _should_ really adopt it.